Funcky/Sequence/Sequence.CycleRange.cs

```diff
@@ -11,15 +13,20 @@ public static partial class Sequence
         [Pure]
         public static IEnumerable<TItem> CycleRange<TItem>(IEnumerable<TItem> sequence)
             where TItem : notnull
-        {
-            var list = sequence.ToList();
+            => CycleSequenceIfNotEmpty(sequence.Materialize());
 
-            if (list.None())
-            {
-                throw new ArgumentException("An empty sequence cannot be cycled", nameof(sequence));
-            }
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static IEnumerable<TItem> CycleSequenceIfNotEmpty<TItem>(IReadOnlyCollection<TItem> sequence)
+            where TItem : notnull
+            => sequence.Count != 0
+                ? ProjectCycles(sequence)
+                : throw new ArgumentException("An empty sequence cannot be cycled", nameof(sequence));
 
-            return Cycle(sequence).SelectMany(Identity);
-        }
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static IEnumerable<TItem> ProjectCycles<TItem>(IReadOnlyCollection<TItem> list)
+            where TItem : notnull
+            => from cycle in Cycle(list)
+               from element in cycle
+               select element;
```

Comment:
Out of curiosity: Why did you change this to linq query syntax?

Reply:
Because we can avoid calling Functional.Identity this way. I do not have a strong pereference.